File: src/main/java/com/example/auth/PasswordEncryptor.java
package com.example.auth;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.security.crypto.password.PasswordEncoder;

@Converter
public class PasswordEncryptor implements AttributeConverter<String, String> {
    private final PasswordEncoder passwordEncoder = new BCryptPasswordEncoder();

    @Override
    public String convertToDatabaseColumn(String attribute) {
        return passwordEncoder.encode(attribute);
    }

    @Override
    public String convertToEntityAttribute(String dbData) {
        return null;
    }

}

File: backend/src/main/kotlin/com/ptah/auth/PasswordEncryptor.kt
package com.ptah.auth

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.security.crypto.password.PasswordEncoder

@Converter
class PasswordEncryptor : AttributeConverter<String?, String> {
    private val passwordEncoder: PasswordEncoder = BCryptPasswordEncoder()
    override fun convertToDatabaseColumn(attribute: String?): String = passwordEncoder.encode(attribute)
    override fun convertToEntityAttribute(dbData: String): String = dbData
}
